File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    
    // 정렬은 필수!
  lost.sort((a, b) => a - b);
  reserve.sort((a, b) => a - b);
  let realLost = lost.filter(x => !reserve.includes(x));
  let realReserve = reserve.filter(x => !lost.includes(x));

  for (let r of realReserve) {
    if (realLost.includes(r - 1)) {
      realLost = realLost.filter(x => x !== r - 1);
    } else if (realLost.includes(r + 1)) {
      realLost = realLost.filter(x => x !== r + 1);
    }
  }

  return n - realLost.length;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost = set(lost)
    reserve = set(reserve)

    # 1️⃣ 여벌도 있고 도난도 당한 학생 제거
    common = lost & reserve
    lost -= common
    reserve -= common

    # 2️⃣ 체육복 빌려주기
    for r in sorted(reserve):
        if r - 1 in lost:
            lost.remove(r - 1)
        elif r + 1 in lost:
            lost.remove(r + 1)

    # 3️⃣ 수업 가능한 학생 수
    return n - len(lost)